contracts/0.8.9/WithdrawalQueue.sol

```diff
@@ -184,56 +187,94 @@ abstract contract WithdrawalQueue is AccessControlEnumerable, PausableUntil, Wit
+            claimableEthValues[i] = _getClaimableEther(_requestIds[i], _hints[i]);
+        }
     }
 
-    /// @notice Claim withdrawals batch once finalized (claimable).
-    /// @param _claimWithdrawalInputs list of withdrawal request ids and hints to claim
-    function claimWithdrawals(ClaimWithdrawalInput[] calldata _claimWithdrawalInputs) external {
-        for (uint256 i = 0; i < _claimWithdrawalInputs.length; ++i) {
-            _claimWithdrawalTo(_claimWithdrawalInputs[i].requestId, _claimWithdrawalInputs[i].hint, msg.sender);
-            _emitTransfer(msg.sender, address(0), _claimWithdrawalInputs[i].requestId);
+    /// @notice Claim a batch of withdrawal requests once finalized (claimable) sending ether to `_recipient`
+    /// @param _requestIds array of request ids to claim
+    /// @param _hints checkpoint hint for each id.
+    ///   Can be retrieved with `findCheckpointHints()` or `findCheckpointHintsUnbounded()`
+    /// @param _recipient address where claimed ether will be sent to
+    /// @dev
+    ///  Reverts if recipient is equal to zero
+    ///  Reverts if any requestId or hint in arguments are not valid
+    ///  Reverts if any request is not finalized or already claimed
+    ///  Reverts if msg sender is not an owner of the requests
+    function claimWithdrawalsTo(uint256[] calldata _requestIds, uint256[] calldata _hints, address _recipient)
+        external
+    {
+        if (_recipient == address(0)) revert ZeroRecipient();
+
+        for (uint256 i = 0; i < _requestIds.length; ++i) {
+            _claim(_requestIds[i], _hints[i], _recipient);
+            _emitTransfer(msg.sender, address(0), _requestIds[i]);
         }
     }
 
-    ///  @notice Claim `_requestId` request and transfer locked ether to the owner
-    ///  @param _requestId request id to claim
-    ///  @param _hint hint for checkpoint index to avoid extensive search over the checkpointHistory.
-    ///   Can be retrieved with `findCheckpointHint()` or `findCheckpointHintUnbounded()`
-    /// @param _recipient address where claimed ether will be sent to
-    function claimWithdrawalTo(uint256 _requestId, uint256 _hint, address _recipient) external {
-        _claimWithdrawalTo(_requestId, _hint, _recipient);
-        _emitTransfer(msg.sender, address(0), _requestId);
+    /// @notice Claim a batch of withdrawal requests once finalized (claimable) sending locked ether to the owner
+    /// @param _requestIds array of request ids to claim
+    /// @param _hints checkpoint hint for each id.
+    ///   Can be retrieved with `findCheckpointHints()` or `findCheckpointHintsUnbounded()`
+    /// @dev
+    ///  Reverts if any requestId or hint in arguments are not valid
+    ///  Reverts if any request is not finalized or already claimed
+    ///  Reverts if msg sender is not an owner of the requests
+    function claimWithdrawals(uint256[] calldata _requestIds, uint256[] calldata _hints) external {
```

Comment:
Yep. We can. But I thought about UX safety here. It seems safer to have a default method that claims to the sender and if you really need it you can use an extended version. Making everyone use an extended version by default can increase the probability of sending funds to a wrong address.

Reply:
I am not against the method, I propose to call another inside

Reply:
Oh. Got you. 

I decided that it's a bit more readable as is, because unwinding a loop doesn't really hurt the readability, but splitting the logic into 3 pieces instead of 2 does. 
Also, public method generates a tiny bit of additional bytecode and cost a tiny bit more gas. So, it's not a strong opinion, but I prefer it like this.